src/sinks/datadog_archives.rs

```diff
@@ -115,43 +140,112 @@ pub struct DatadogArchivesSinkConfig {
+    /// [canned_acl]: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#canned-acl
+    pub acl: Option<S3CannedAcl>,
+
+    /// Grants `READ`, `READ_ACP`, and `WRITE_ACP` permissions on the created objects to the named [grantee].
+    ///
+    /// This allows the grantee to read the created objects and their metadata, as well as read and
+    /// modify the ACL on the created objects.
+    ///
+    /// [grantee]: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#specifying-grantee
+    pub grant_full_control: Option<String>,
+
+    /// Grants `READ` permissions on the created objects to the named [grantee].
+    ///
+    /// This allows the grantee to read the created objects and their metadata.
+    ///
+    /// [grantee]: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#specifying-grantee
+    pub grant_read: Option<String>,
+
+    /// Grants `READ_ACP` permissions on the created objects to the named [grantee].
+    ///
+    /// This allows the grantee to read the ACL on the created objects.
+    ///
+    /// [grantee]: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#specifying-grantee
+    pub grant_read_acp: Option<String>,
+
+    /// Grants `WRITE_ACP` permissions on the created objects to the named [grantee].
+    ///
+    /// This allows the grantee to modify the ACL on the created objects.
+    ///
+    /// [grantee]: https://docs.aws.amazon.com/AmazonS3/latest/dev/acl-overview.html#specifying-grantee
+    pub grant_write_acp: Option<String>,
+
+    /// The Server-side Encryption algorithm used when storing these objects.
+    pub server_side_encryption: Option<S3ServerSideEncryption>,
+
+    /// Specifies the ID of the AWS Key Management Service (AWS KMS) symmetrical customer managed
+    /// customer master key (CMK) that will used for the created objects.
+    ///
+    /// Only applies when `server_side_encryption` is configured to use KMS.
+    ///
+    /// If not specified, Amazon S3 uses the AWS managed CMK in AWS to protect the data.
+    pub ssekms_key_id: Option<String>,
+
+    /// The storage class for the created objects.
+    ///
+    /// For more information, see [Using Amazon S3 storage classes][storage_classes].
+    ///
+    /// [storage_classes]: https://docs.aws.amazon.com/AmazonS3/latest/dev/storage-class-intro.html
+    pub storage_class: Option<S3StorageClass>,
+
+    /// The tag-set for the object.
+    pub tags: Option<BTreeMap<String, String>>,
 }
 
-#[derive(Deserialize, Serialize, Default, Debug, Clone)]
+/// ABS-specific configuration options.
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
 #[serde(deny_unknown_fields)]
 pub struct AzureBlobConfig {
+    /// The Azure Blob Storage Account connection string.
+    ///
+    /// Authentication with access key is the only supported authentication method.
     pub connection_string: String,
 }
 
-#[derive(Deserialize, Serialize, Default, Debug, Clone)]
-#[serde(deny_unknown_fields)]
```

Comment:
Same as above, was this intentionally dropped?

Reply:
Intentionally, yes, but erroneously. I'll fix that.

Reply:
Fixed in 18c22d4b8.